lib/rmt/cli/smt_importer.rb

```diff
@@ -0,0 +1,163 @@
+        repo = Repository.create!(external_url: repo_url, name: repo_name, local_path: local_path)
+      end
+
+      if product
+        assoc = RepositoriesServicesAssociation.find_by(service: product.service, repository: repo)
+        unless assoc
+          RepositoriesServicesAssociation.create!(service: product.service, repository: repo)
+          puts "Added association between #{repo.name} and product #{product_id}"
+        end
+      else
+        warn "Product #{product_id} not found"
+      end
+    end
+  end
+
+  def import_systems
+    read_csv('systems').each do |row|
+      login, password, hostname, registered_at = row
+
+      next unless System.find_by(login: login, password: password).nil?
+
+      # rubocop:disable Rails/TimeZone
+      System.create!(
+        login: login,
+        password: password,
+        hostname: hostname,
+        registered_at: Time.at(registered_at.to_i)
+      )
+      # rubocop:enable Rails/TimeZone
+      puts "Imported system #{login}"
+    end
+  end
+
+  def import_activations
+    read_csv('activations').each do |row|
+      login, product_id = row
+
+      product = Product.find_by(id: product_id)
+      system = System.find_by(login: login)
+
+      if !system
+        warn "System #{login} not found"
+        next
+      elsif !product
+        warn "Product #{product_id} not found"
+        next
+      else
+        activation = Activation.find_by(system: system, service: product.service)
+        unless activation
+          Activation.create!(system: system, service: product.service)
+          puts "Imported activation of #{product_id} for #{login}"
+        end
+      end
+    end
+  end
+
+  def import_hardware_info
+    systems = {}
+
+    # create hardware infos
+    read_csv('hw_info').each do |row|
+      login, key, value = row
+      next unless key == 'machinedata'
+
+      systems[login] = JSON.parse(value)['hwinfo']
+    end
+
+    systems.each do |login, info|
+      system = System.find_by(login: login)
+
+      unless system
+        warn "System #{login} not found"
+        next
+      end
+      info.delete('hostname')
+
+      HwInfo.find_or_initialize_by(system: system).update!(info)
+      puts "Hardware information stored for system #{login}"
+    end
+  end
+
+  def run(argv)
+    parse_cli_arguments argv
+    check_products_exist
+
+    ActiveRecord::Base.transaction do
+      import_repositories
+      import_custom_repositories
+    end
+
+    raise ImportException if config.no_systems
+
+    ActiveRecord::Base.transaction do
+      import_systems
+      import_activations
+      import_hardware_info
+    end
+  end
+
+  def parse_cli_arguments(argv)
+    parser = OptionParser.new do |parser|
+      parser.on('-d', '--data PATH', 'Path to unpacked SMT data tarball') { |path| config.data_dir = path }
+      parser.on('--no-systems', 'Import no systems to rmt')               { config.no_systems = true }
```

Comment:
'Import no systems to rmt' =>  "Don't import the systems that were registered to the SMT"